src/lomc.jl

```diff
@@ -59,29 +64,37 @@ Can be advanced a step forward with [`advance!`](@ref).
 * `pv`: vector from the previous step.
 * `wm`: working memory.
 * `pnorm`: previous walker number (see [`walkernumber`](@ref)).
-* `params`: the [`FciqmcRunStrategy`](@ref).
+* `s_strat`: shift strategy.
+* `τ_strat`: time step strategy.
+* `shift`: shift to control the walker number.
+* `dτ`: time step size.
 * `id`: string ID appended to reported column names.
 
 See also [`QMCState`](@ref), [`ReplicaStrategy`](@ref), [`replica_stats`](@ref),
 [`lomc!`](@ref).
 """
-mutable struct ReplicaState{H,T,V,W,R<:FciqmcRunStrategy{T}}
+mutable struct ReplicaState{H,V,W,SS<:ShiftStrategy,TS<:TimeStepStrategy,SP}
     # Future TODO: rename these fields, add interface for accessing them.
     hamiltonian::H
     v::V       # vector
     pv::V      # previous vector.
     wm::W      # working memory. Maybe working memories could be shared among replicas?
-    pnorm::T   # previous walker number - used to control the shift
-    params::R  # params: step, laststep, dτ...
+    s_strat::SS # shift strategy
+    τ_strat::TS # time step strategy
+    shift_parameters::SP # norm, shift, time_step; is mutable
     id::String # id is appended to column names
 end
 
-function ReplicaState(h, v, wm, params, id="")
+function ReplicaState(h, v, wm, s_strat, τ_strat, shift, dτ::Float64, id="")
     if isnothing(wm)
         wm = similar(v)
     end
     pv = zerovector(v)
-    return ReplicaState(h, v, pv, wm, walkernumber(v), params, id)
+    sp = initialise_shift_parameters(s_strat, shift, walkernumber(v), dτ)
+    return ReplicaState(h, v, pv, wm, s_strat, τ_strat, sp, id)
+    # return ReplicaState{
+    #     typeof(h),typeof(v),typeof(wm),typeof(s_strat),typeof(τ_strat),typeof(sp)
+    # }(h, v, pv, wm, s_strat, τ_strat, sp, id)
 end
```

Comment:
This method is not called when a `QMCState` is created.

Reply:
I'm not sure I understand this comment. The method is called 391 times during CI testing according to Coveralls.